### PR TITLE
[opentitantool] Make the verilator startup time configurable 

### PR DESCRIPTION
**Pull request by cfrantz**
1. Spawn a thread to consume and log the verilator stdout.
2. Add a `--verilator_timeout` parameter to control how long to wait
   for the verilator subprocess to start.  Set the default value to
   60 seconds.

Signed-off-by: Chris Frantz <cfrantz@google.com>

### CODE REVIEW

**Review by milesdai — APPROVED** [review]
Thanks for getting this out so fast!

**Comment by rswarbrick** [comment]
If this is verilator-specific, I'd suggest just never timing out. It's not like there's some flaky physical connection that might have gone down. If you really think that the feature might be useful, please can you set the default timeout to "no timeout"?

File: sw/host/opentitanlib/src/transport/verilator/stdout.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+use std::io::Read;
+use std::process::ChildStdout;
+use std::sync::{Arc, Mutex};
+
+// The whole point of this module being named `stdout.rs` is so that the log message
+// looks like `[<timestamp> INFO ...verilator::stdout] <message>`.
+
+// Accumulates output from verilator's `stdout`.
+pub fn accumulate(stdout: ChildStdout, accumulator: Arc<Mutex<String>>) {
+    match worker(stdout, accumulator) {
```
Comment (mcy) [inline sw/host/opentitanlib/src/transport/verilator/stdout.rs:15]: `if let` might be better here.

**Review by mcy — APPROVED** [review]

**Comment by mcy** [comment]
Rupert: please provide a reason why you think this is a good default, if you could. This change is necessary to improve CI failure latency.

**Comment by cfrantz** [comment]
> If this is verilator-specific, I'd suggest just never timing out. It's not like there's some flaky physical connection that might have gone down. If you really think that the feature might be useful, please can you set the default timeout to "no timeout"?

You're not wrong, and yet, this seems to be one of the problems we're dealing with in CI: the verilator subprocess just sits there indolently.  If we determine that 60s is not enough time for verilator to get started, we'll adjust the timeout.  Considering that bazel enforces a timeout of 5m for short timeouts, 60s seems like a reasonable timeout for letting verilator start.

File: sw/host/opentitanlib/src/transport/verilator/stdout.rs (hunk 1 above)
Comment (cfrantz) [reply sw/host/opentitanlib/src/transport/verilator/stdout.rs]: Done.

**Pull request merged**